bundles/org.openhab.binding.intesis/src/main/java/org/openhab/binding/intesis/internal/handler/IntesisHomeHandler.java

```diff
@@ -0,0 +1,529 @@
+                                break;
+                        }
+                    }
+                }
+            } catch (JsonSyntaxException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+            }
+        }
+    }
+
+    public void login() {
+        String contentString = "{\"command\":\"login\",\"data\":{\"username\":\"Admin\",\"password\":\"" + password
+                + "\"}}";
+        String response = api.postRequest(ipAddress, contentString);
+        if (response != null) {
+            try {
+                Response resp = gson.fromJson(response, Response.class);
+                boolean success = resp.success;
+                if (success) {
+                    Data data = gson.fromJson(resp.data.toString(), Data.class);
+                    Id id = gson.fromJson(data.id.toString(), Id.class);
+                    if (id.sessionID.toString() != null && !id.sessionID.toString().isEmpty()) {
+                        sessionId = id.sessionID.toString();
+                        updateStatus(ThingStatus.ONLINE);
+                    } else {
+                        updateStatus(ThingStatus.OFFLINE);
+                    }
+                }
+            } catch (JsonSyntaxException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+            }
+        }
+    }
+
+    public @Nullable String logout() {
+        String contentString = "{\"command\":\"logout\",\"data\":{\"sessionID\":\"" + sessionId + "\"}}";
+        String response = api.postRequest(ipAddress, contentString);
+        return response;
+    }
+
+    public void populateProperties() {
+        String contentString = "{\"command\":\"getinfo\",\"data\":\"\"}";
+        String response = api.postRequest(ipAddress, contentString);
+        logger.trace("getInfo response : {}", response);
+        if (response != null) {
+            try {
+                Response resp = gson.fromJson(response, Response.class);
+                boolean success = resp.success;
+                if (success) {
+                    Data data = gson.fromJson(resp.data.toString(), Data.class);
+                    Info info = gson.fromJson(data.info.toString(), Info.class);
+                    properties.put(PROPERTY_VENDOR, "Intesis");
+                    properties.put(PROPERTY_MODEL_ID, info.deviceModel);
+                    properties.put(PROPERTY_SERIAL_NUMBER, info.sn);
+                    properties.put(PROPERTY_FIRMWARE_VERSION, info.fwVersion);
+                    properties.put(PROPERTY_MAC_ADDRESS, info.wlanSTAMAC);
+                    updateProperties(properties);
+                    updateStatus(ThingStatus.ONLINE);
+                }
+            } catch (JsonSyntaxException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+            }
+        }
+    }
+
+    public void getDataPoints() {
```

Comment:
Just like `populateProperties` you should probably rename this method to `populateChannels`

Reply:
Renamed